### PR TITLE
Fix the ESM modules to properly support Node 12.16

### PR DESCRIPTION
**Pull request by blittle**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,9 +23,9 @@ To set up the sample application:
 2. Change into the `sfcc-sample-apps` folder:
 `cd sfcc-sample-apps`
 
-3. Copy the `api.example.js` file located at `/packages/storefront-lwc/scripts/`, save it as `api.js`, and make sure `api.js` is added to your `.gitignore` file.
+3. Copy the `api.example.mjs` file located at `/packages/storefront-lwc/scripts/`, save it as `api.mjs`, and make sure `api.mjs` is added to your `.gitignore` file.
```
Comment (blittle) [inline README.md]: Currently the dev build is broken from a error swallowed by `lwc-services`. It relates to our native module usage with Node 12.16. `lwc-services` doesn't support esm modules very well. I'm going to try to fix that, but until then, it is easier to keep our native module support by switching these entry files to `.mjs`.

File: packages/storefront-lwc/lwc-services.config.js [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-  server: {
-    customConfig: "./scripts/sample-app.js"
-  },
-  devServer: {
-    proxy: {
-      "/api": "http://localhost:3002",
-      "/assets": "http://localhost:3002",
-      "/manifest.json": "http://localhost:3002",
-      "/apiconfig.js": "http://localhost:3002"
+    buildDir: './dist/public',
```
Comment (blittle) [inline packages/storefront-lwc/lwc-services.config.js]: Prettier ran on this file

File: packages/storefront-lwc/package.json [hunk 3]
```diff
@@ -3,7 +3,6 @@
   "version": "0.0.1",
   "description": "Sample Storefront Application with Lightning Web Components",
   "license": "MIT",
-  "type": "module",
```
Comment (blittle) [inline packages/storefront-lwc/package.json]: In addition to renaming those files to `.mjs` extensions, I removed the default to native modules. This means only files with `.mjs` extension will be resolved to native. That's only relevant to files in the `scripts` directory. All the core and bff code still keep `type: "module"`

File: packages/storefront-lwc/scripts/runtime.mjs [hunk 4]
```diff
@@ -27,6 +27,28 @@ const publicDir = `${templateDir}/dist/public/`;
...
 
+function validateConfig(config) {
+    const REQUIRED_KEYS = [
+        'COMMERCE_API_PATH',
+        'COMMERCE_CLIENT_API_SITE_ID',
+        'COMMERCE_CLIENT_CLIENT_ID',
+        'COMMERCE_CLIENT_REALM_ID',
+        'COMMERCE_CLIENT_INSTANCE_ID',
+        'COMMERCE_CLIENT_ORGANIZATION_ID',
+        'COMMERCE_CLIENT_SHORT_CODE',
```
Comment (blittle) [inline packages/storefront-lwc/scripts/runtime.mjs]: I added some validation of the config file / env variables. If one of these is not present, the app will tell you its needed and exit.

File: packages/storefront-lwc/scripts/runtime.mjs [hunk 5]
```diff
@@ -57,18 +81,17 @@ const mode = process.env.NODE_ENV || 'development';
 
     // start the server
     const server = sampleApp.expressApplication.listen(port, () => {
+        const portToTellUser =
+            process.env.SFCC_DEV_MODE === 'true' ? 3000 : server.address().port;
```
Comment (blittle) [inline packages/storefront-lwc/scripts/runtime.mjs]: Tell the user the port the app is on based upon dev/prod mode. Dev mode it is always 3000, while prod is according to the env variable.

File: packages/storefront-lwc/lwc-services.config.js [hunk 6]
```diff
@@ -1,21 +1,21 @@
...
+    resources: [],
+    sourceDir: './src',
+    moduleDir: './src/modules',
+    server: {
+        customConfig: './server/sample-app.js',
```
Comment (syedshehrozhussain) [inline packages/storefront-lwc/lwc-services.config.js]: does it matter if sample-app.js does not have .mjs explicitly?

File: packages/storefront-lwc/scripts/runtime.mjs [hunk 7]
```diff
@@ -27,6 +27,28 @@ const publicDir = `${templateDir}/dist/public/`;
...
+        'COMMERCE_CLIENT_REALM_ID',
+        'COMMERCE_CLIENT_INSTANCE_ID',
+        'COMMERCE_CLIENT_ORGANIZATION_ID',
+        'COMMERCE_CLIENT_SHORT_CODE',
+    ];
+
+    REQUIRED_KEYS.forEach(KEY => {
+        if (!config[KEY]) {
+            console.log(
+                `Make sure ${KEY} is defined within api.js or as an environment variable`
```
Comment (zsardone) [inline packages/storefront-lwc/scripts/runtime.mjs]: api.mjs?

File: packages/storefront-lwc/lwc-services.config.js (hunk 6 above)
Comment (zsardone) [reply packages/storefront-lwc/lwc-services.config.js]: this should be change to sample-app.mjs

File: packages/storefront-lwc/lwc-services.config.js (hunk 2 above)
Comment (zsardone) [reply packages/storefront-lwc/lwc-services.config.js]: does this file need to be .mjs?

Comment (blittle) [reply packages/storefront-lwc/lwc-services.config.js]: Nope :)

**Pull request merged**